sdk/keyvault/azure-security-keyvault/tests/test_keys_async.py

```diff
@@ -0,0 +1,372 @@
+        # validate all our deleted keys are returned by list_deleted_keys
+        result = client.list_deleted_keys()
+        await self._validate_key_list(result, expected)
+
+    @ResourceGroupPreparer()
+    @VaultClientPreparer()
+    @await_prepared_test
+    async def test_backup_restore(self, vault_client, **kwargs):
+        self.assertIsNotNone(vault_client)
+        client = vault_client.keys
+        key_name = self.get_resource_name("keybak")
+        key_type = "RSA"
+
+        # create key
+        created_bundle = await client.create_key(key_name, key_type)
+
+        # backup key
+        key_backup = await client.backup_key(created_bundle.name)
+        self.assertIsNotNone(key_backup, "key_backup")
+
+        # delete key
+        await client.delete_key(created_bundle.name)
+        # can add test case to see if we do get_deleted should return error
+
+        # restore key
+        restored = await client.restore_key(key_backup)
+        self.assertEqual(created_bundle.id, restored.id)
+        self._assert_key_attributes_equal(created_bundle, restored)
+
+    @ResourceGroupPreparer()
+    @VaultClientPreparer(enable_soft_delete=True)
+    @await_prepared_test
+    async def test_recover_purge(self, vault_client, **kwargs):
+
+        self.assertIsNotNone(vault_client)
+        client = vault_client.keys
+        keys = {}
+
+        # create keys to recover
+        for i in range(self.list_test_size):
+            key_name = self.get_resource_name("keyrec{}".format(i))
+            keys[key_name] = await client.create_key(key_name, "RSA")
+
+        # create keys to purge
+        for i in range(self.list_test_size):
+            key_name = self.get_resource_name("keyprg{}".format(i))
+            keys[key_name] = await client.create_key(key_name, "RSA")
+
+        # delete all keys
+        for key_name in keys.keys():
+            await client.delete_key(key_name)
+
+        await self._poll_until_resource_found(client.get_deleted_key, keys.keys())
+
+        # recover select keys
+        for key_name in [s for s in keys.keys() if s.startswith("keyrec")]:
+            recovered_key = await client.recover_deleted_key(key_name)
+            expected_key = keys[key_name]
+            self._assert_key_attributes_equal(expected_key, recovered_key)
+
+        # purge select keys
+        for key_name in [s for s in keys.keys() if s.startswith("keyprg")]:
+            await client.purge_deleted_key(key_name)
+
+        # validate the recovered keys
+        expected = {k: v for k, v in keys.items() if k.startswith("keyrec")}
+        await self._poll_until_resource_found(client.get_key, expected.keys())
+
+        actual = {}
+        for k in expected.keys():
+            actual[k] = await client.get_key(k)
+
+        self.assertEqual(len(set(expected.keys()) & set(actual.keys())), len(expected))
+
+    @ResourceGroupPreparer()
+    @VaultClientPreparer()
+    async def test_key_wrap_and_unwrap(self, vault_client, **kwargs):
```

Comment:
This test is broken:
- Doesn't execute because it's missing the await decorator
- Can't run offline because it has no recording (likely cause of the `getaddrinfo` error you were seeing)
- Raises because on this branch `KeyVaultTestCase` doesn't have a `plain_text` attribute